openassessment/xblock/openassessmentblock.py

```diff
@@ -329,6 +335,10 @@ def _create_ui_models(self):
         """
         ui_models = [UI_MODELS["submission"]]
         for assessment in self.valid_assessments:
+            if assessment["name"] == "staff-assessment" and assessment["required"] == False:
+                # Check if staff have graded the assessment
+                # else
+                continue
```

Comment:
So if the staff assessment step is optional and staff have not given an assessment, we do not show the step in the UI.